aiohttp/http_parser.py

```diff
@@ -727,30 +727,35 @@ def flush(self) -> bytes:
             self.decompressor = BrotliDecoder()  # type: Any
         else:
             zlib_mode = (16 + zlib.MAX_WBITS
-                         if encoding == 'gzip' else -zlib.MAX_WBITS)
+                         if encoding == 'gzip' else zlib.MAX_WBITS)
             self.decompressor = zlib.decompressobj(wbits=zlib_mode)
 
     def set_exception(self, exc: BaseException) -> None:
         self.out.set_exception(exc)
 
     def feed_data(self, chunk: bytes, size: int) -> None:
+        if not size:
+            return
+
         self.size += size
+
+        # RFC1950
+        # bits 0..3 = CM = 0b1000 = 8 = "deflate"
+        # bits 4..7 = CINFO = 1..7 = windows size.
+        if not self._started_decoding and self.encoding == 'deflate' and chunk[0] & 0xf != 8:
```

Comment:
on very small chunks it's not enough to trigger error in the first call to decompressor. I wrote the test on it.

Reply:
I did not check, but this code might be slightly slower. Actually, such a case can happen rarely, since I can't imagine a real situation where compressed chunk (went out of compressor) might be split in such a weird way.